staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/controller.go

```diff
@@ -326,6 +326,12 @@ func (c *celAdmissionController) Validate(
 				}
 			}
 
+			namespace, err := c.policyController.matcher.GetNamespace(a.GetNamespace())
+			// if it is cluster scoped, namespace will be null
+			if err != nil {
+				namespace = nil
+			}
```

Comment:
Can we avoid relying on an error to check if a namespace is cluster scoped?  If `a.GetNamespace()` is empty can we be guaranteed the object is cluster scoped?

I'm concerned that there might be cases where a namespace scoped object results in an error here and we end up mis-handling the case because we blindly interpreted any error to mean that the object was cluster scoped.  I would rather we do a clean "has namespace" check and then report any errors when we call `matcher.GetNamespace()` using normal error handling.

Reply:
`a.GetNamespace()` returns empty string for cluster-scoped resources. I've added check to see if the namespace name is empty.